### PR TITLE
Core/Scripts Yogg-Saron Portals

### LINKED ISSUE #16390: [3.3.5][Ulduar] Core/Gameobject Yogg-Saron Portals
Clicking them causes the portal to despawn, but you're not actually ported.

Related spellclick spells: 63989, 63997, 63998

Portals: 34072

Rev https://github.com/TrinityCore/TrinityCore/commit/e6a7818a9a12b085c8ea07cfb8b3c6d1d700a112

### PR DESCRIPTION
**Pull request by sirikfoll**
**Changes proposed**:
- Update and Moves old db scripts(non-functional anymore and a little bit wrong) of the npc http://wotlk.openwow.com/npc=34072 to core
- Use right npc_spellclick spell

**Target branch(es)**: 335/6x

**Issues addressed**: Fixes #16390

**Tests performed**: Built and tested

**Known issues and TODO list**: None

### CODE REVIEW

**Comment by joschiwald** [comment]
can you confirm spellclick spell with sniffs?

**Comment by sirikfoll** [comment]
Yes, `[11] UNIT_FIELD_INTERACT_SPELLID: 51347/7.195247E-41`
sniff available at https://community.trinitycore.org/topic/12401-ulduar-10-with-hard-modes/

**Comment by sirikfoll** [comment]
"comment after DATA_ILLUSION in ulduar.h and the static_assert in instance_ulduar.cpp" were already removed somewhere.

Changed `cast_flags` to 3
```
ServerToClient: SMSG_SPELL_GO (0x046E) Length: 103 ConnIdx: 1 Time: 07/21/2015 02:40:08.703 Number: 238496
(Cast) CasterGUID: Full: 0x08322000000000000000000006C2DA00 Player/0 R3208/S0 Map: 0 (Eastern Kingdoms) Low: 113433088
(Cast) CasterUnit: Full: 0x08322000000000000000000006C2DA00 Player/0 R3208/S0 Map: 0 (Eastern Kingdoms) Low: 113433088
(Cast) CastID: 0
(Cast) SpellID: Teleport Visual Only (51347)
(Cast) SpellXSpellVisualID: 19714
(Cast) CastFlags: 17040128
(Cast) CastTime: 2927324536
(Cast) HitTargetsCount: 1
(Cast) MissTargetsCount: 0
(Cast) MissStatusCount: 0
(Cast) (Target) Flags: Unit (2)
(Cast) (Target) HasSrcLocation: False
(Cast) (Target) HasDstLocation: False
(Cast) (Target) HasOrientation: False
(Cast) (Target) Unit: Full: 0x0
(Cast) (Target) Item: Full: 0x0
(Cast) (Target) Name: 
(Cast) RemainingPowerCount: 0
(Cast) (MissileTrajectory) TravelTime: 0
(Cast) (MissileTrajectory) Pitch: 0
(Cast) (Ammo) DisplayID: 0
(Cast) (Ammo) InventoryType: NonEquip (0)
(Cast) DestLocSpellCastIndex: 0
(Cast) TargetPointsCount: 0
(Cast) (Immunities) School: 0
(Cast) (Immunities) Value: 0
(Cast) (Predict) Points: 0
(Cast) (Predict) Type: 0
(Cast) (Predict) BeaconGUID: Full: 0x0
(Cast) [0] HitTarget: Full: 0x08322000000000000000000006C2DA00 Player/0 R3208/S0 Map: 0 (Eastern Kingdoms) Low: 113433088
(Cast) CastFlagsEx: 16
(Cast) HasRuneData: False

**Comment by joschiwald** [comment]
looks valid

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_yogg_saron.cpp [hunk 1]
```diff
@@ -1419,7 +1430,9 @@ class npc_descend_into_madness : public CreatureScript
             {
                 if (!result)
                     return;
+
                 clicker->RemoveAurasDueToSpell(SPELL_BRAIN_LINK);
+                DoCast(clicker, IllusionSpells[_instance->GetData(DATA_ILLUSION)], true);
```
Comment (joschiwald) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_yogg_saron.cpp]: is it possible that GetData(DATA_ILLUSION) returns a value not 0..2?

Comment (sirikfoll) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_yogg_saron.cpp]: afaik only if you use SetData do change it, other places where this GetData is used does not perform any check too, and even if return a weird value(which should not happen), DoCast do check already checks if the spell passed is valid.

**Comment by L30m4nc3r** [comment]
@sirikfoll thanks! Work Perfect!
But have 1 problem. After raid wipe event not reseted.

**Comment by Rushor** [comment]
this issue has nothing to fo with encounterreset

**Comment by L30m4nc3r** [comment]
@Rushor okay i open issue?

**Comment by sirikfoll** [comment]
yeah, i saw that too, but happens without this PR too, the trigger(Voice of Yogg-Saron) never evades

**Comment by L30m4nc3r** [comment]
@Rushor okay thanks. #16723

**Pull request closed without merging**